pkg/log/auditlog/auditlog.go

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (C) 2024 IOTech Ltd
+//
+
+package auditlog
+
+import (
+	"io"
+	"log/slog"
+	"os"
+	"time"
+)
+
+type logger struct {
+	logger *slog.Logger
+}
+
+// AuditLog builds up an audit log entry with the following fields
+type AuditLog struct {
+	// actor indicates the identity of the user or system that initiated the action
+	actor string
+	// action indicates the type of action that was performed
+	action ActionType
+	// description contains a brief description of the action
+	description string
+	// details contains extra information regarding the action
+	details any
+	// severity indicates the severity of the action
+	severity string
+	// timestamp indicates the time when the action was performed
+	timestamp time.Time
+}
+
+// InitLogger creates an instance of Logger
+func InitLogger(logWriter io.Writer) Logger {
+	if logWriter == nil {
+		logWriter = os.Stdout
+	}
+
+	l := slog.New(slog.NewTextHandler(logWriter, nil))
+	return &logger{logger: l}
+}
+
+// Add adds an audit log entry to the log writer
+func (l *logger) Add(actor string, action ActionType, description string, details any, severity Severity, timestamp time.Time) {
```

Comment:
apply the coverage level, `Base`, `Advanced`, and `Full`
use `LogBase`, `LogAdvanced`, and `LogFull` instead of `Add`
also add `LogBasef`, `LogAdvancedf`, and `LogFullf`

Reply:
Just had a thought that perhaps we don't need these methods with formatting functionality because the audit log needs to record information as key-value pairs rather than consolidating them into a single message.